src/nodes.coffee

```diff
@@ -1728,7 +1759,8 @@ exports.Assign = class Assign extends Base
         # destructured variables.
         @variable.base.lhs = yes
         return @compileDestructuring o unless @variable.isAssignable()
-
+        # Object destructuring. Can be removed once ES proposal hits Stage 4.
+        return @compileObjectDestruct(o) if @variable.isObject() and @variable.hasSplat()
```

Comment:
Gonna moan about this again (sorry) - I'd rather not make the `hasSplat` helper on `Value` when it's only used here, and `@variable.base.hasSplat()` would work just as well.